src/main/java/cloud/orbit/actors/cluster/state/RedisClusterTracker.java

```diff
@@ -161,6 +162,29 @@ public boolean isThisNodeDead()
         return isThisNodeInState(NodeState.PRESUMED_DEAD);
     }
 
+    public boolean isNodeRunningOrStopping()
+    {
+        final NodeState nodeState = this.getNodeState();
+        return nodeState == NodeState.RUNNING || nodeState == NodeState.STOPPING;
+    }
+
+    public boolean hasLostConnection()
+    {
+        return hasLostConnection;
+    }
+
+    /**
+     * Attempts to forcefully shutdown the JVM,
+     * as there is no graceful way to handle a lost
+     * connection to Redis.
+     */
+    public void handleLostConnection()
+    {
+        logger.error("FATAL: Node {} detected itself as dead. Exiting to restart.", localAddress);
+        hasLostConnection = true;
```

Comment:
Should we use AtomicBoolean compareAndSet?

Reply:
I can look into that, never used it before, but that makes sense to me 👍

Reply:
worked just fine, made the change